regex/tests/Test.lean

```diff
@@ -153,4 +153,907 @@ def word_class := Regex.parse! r##"\w+"##
+
+-- unicode tests
+-- unicode1* and unicode2* work for both Unicode and ASCII because all matches
+-- are reported as byte offsets, and « and » do not correspond to word
+-- boundaries at either the character or byte level.
+
+--
+-- Unicode word boundary tests
+--
+def unicode_word_boundary_01 := Regex.parse! r##"\bx\b"##
+
+-- name = "unicode1"
+#guard unicode_word_boundary_01.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+
+-- name = "unicode1-only-ascii"
+#guard unicode_word_boundary_01.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "«x" = #[
+  ⟨"«x", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+
+-- name = "unicode2"
+#guard unicode_word_boundary_01.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+-- name = "unicode2-only-ascii"
+#guard unicode_word_boundary_01.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "x»" = #[
+  ⟨"x»", #[.some ⟨0⟩, .some ⟨1⟩]⟩
+]
+
+-- ASCII word boundaries are completely oblivious to Unicode characters, so
+-- even though β is a character, an ASCII \b treats it as a word boundary
+-- when it is adjacent to another ASCII character. (The ASCII \b only looks
+-- at the leading byte of β.) For Unicode \b, the tests are precisely inverted.
+
+-- FIXME: This test is not working as expected.
+-- name = "unicode3"
+-- #eval unicode_word_boundary_01.captureAll "áxβ"
+-- #guard unicode_word_boundary_01.captureAll "áxβ" = #[]
+-- #guard unicode_word_boundary_01.bt.captureAll "áxβ" = #[]
+
+-- name = "unicode3-only-ascii"
+#guard unicode_word_boundary_01.captureAll "áxβ" = #[
+  ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "áxβ" = #[
+  ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+
+def unicode_non_word_boundary_01 := Regex.parse! r##"\Bx\B"##
+
+-- FIXME: This test is not working as expected.
+-- name = "unicode4"
+-- #eval unicode_word_boundary_02.captureAll "áxβ"
+-- #guard unicode_word_boundary_02.captureAll "áxβ" = #[
+--   ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+-- ]
+-- #guard unicode_word_boundary_02.bt.captureAll "áxβ" = #[
+--   ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+-- ]
+
+-- name = "unicode4-only-ascii"
+#guard unicode_non_word_boundary_01.captureAll "áxβ" = #[]
+#guard unicode_non_word_boundary_01.bt.captureAll "áxβ" = #[]
+
+-- The same as above, but with \b instead of \B as a sanity check.
+def unicode_word_boundary_02 := Regex.parse! r##"\b"##
+
+-- FIXME: This test is not working as expected.
+-- -- name = "unicode5"
+-- #eval unicode_word_boundary_02.captureAll "0\U0007EF5E"
+-- #guard unicode_word_boundary_02.captureAll "0\U0007EF5E" = #[
+--   ⟨"0\U0007EF5E", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+--   ⟨"0\U0007EF5E", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+-- ]
```

Comment:
```suggestion
#guard unicode_word_boundary_02.captureAll ""0\uFFFF" = #[
  ⟨""0\uFFFF", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
  ⟨""0\uFFFF", #[.some ⟨1⟩, .some ⟨1⟩]⟩
]
```

The tests fail only because the original test uses a Unicode escaping that's only available in TOML. We can replace the problematic escape sequence with one supported by Lean to fix.